# Convert inputdata into phyloseq object

*r Biocpkg("phyloseq")* [@mcmurdie2013phyloseq] is the most popular [Biocondcutor](https://bioconductor.org/) package used by the microbiome research field, and `phyloseq-class` objects are a great data-standard for microbiota data in R. Therefore, the core functions in  `XMAS` take `phyloseq-class` object as input. In the phyloseq object, information on OTU abundances, taxonomy of OTUs, the phylogenetic tree and metadata is stored.


This tutorial will introduce you the basic steps to convert results from the in-house pipeline into *phyloseq-class object*. More importantly on how to look at your data and filter appropriately. We will use the inputs from **/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/**.


**Loading packages**
```{r, echo=TRUE, results="hide", warning=FALSE, message=FALSE, fig.align="center"}
library(XMAS2)
library(dplyr)
library(tibble)
library(phyloseq)
```


## DADA2 

**dada2** results from standardized_analytics_workflow_R_function

### Importing results from dada2 pipeline

1. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds

2. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk

3. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt

```{r, warning=FALSE, message=FALSE}
dada2_res <- readRDS(
  system.file(
    "extdata", "dada2_res.rds",
    package = "XMAS2"    
    )
)

sam_tab <- read.table(
    system.file(
        "extdata", "dada2_metadata.tsv",
        package = "XMAS2"
    ),
    sep = "\t",
    header = TRUE,
    stringsAsFactors = FALSE
)

tree <- read_tree(
  system.file(
    "extdata", "tree.nwk",
    package = "XMAS2"    
    )
)
```

### taxa table 

We use `import_dada2_taxa` to convert dada2_res$tax_tab into our own taxa table
```{r, warning=FALSE, message=FALSE}
tax_tab <- import_dada2_taxa(dada2_res$tax_tab)

head(tax_tab, 1)
```


### otu table
```{r, warning=FALSE, message=FALSE}
otu_tab <- dada2_res$seq_tab
# Shouldn't use the Total Number as SampleID (wrong: 123456; right: X123456)
rownames(otu_tab) <- paste0("S", rownames(otu_tab))

head(otu_tab[, 1, F])
```


### metadata table
```{r, warning=FALSE, message=FALSE}
sam_tab <- sam_tab %>% tibble::column_to_rownames("seqID")
# Shouldn't use the Total Number as SampleID (wrong: 123456; right: X123456)
rownames(sam_tab) <- paste0("S", rownames(sam_tab))

head(sam_tab)
```


### phyloseq object
```{r, warning=FALSE, message=FALSE}
dada2_ps <- import_dada2(seq_tab = otu_tab, 
                   tax_tab = tax_tab, 
                   sam_tab = sam_tab, 
                   phy_tree = tree)
dada2_ps
```

we obtain the **phyloseq-class object** and then use it to perform data analysis.

Here, the phyloseq object comprises five components (*OTU Table*, *Sample Data*, *Taxonomy Table*, *Phylogenetic Tree* and *DNAStringSet*).


* otu_table
```{r, warning=FALSE, message=FALSE}
dada2_ps@otu_table %>% data.frame() %>% head()
```

* tax_table
```{r, warning=FALSE, message=FALSE}
dada2_ps@tax_table %>% data.frame() %>% head()
```

* sample_table
```{r, warning=FALSE, message=FALSE}
dada2_ps@sam_data %>% data.frame() %>% head()
```


##  Summarize phyloseq-class object
```{r, warning=FALSE, collapse=TRUE}
summarize_phyloseq(dada2_ps)
```

The minus account of the OTU counts is *51181* in the phyloseq object, and we can use it as the threshold to rarefy. 

Notice the **Sparsity (0.865)**, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.

## Systematic Information
```{r}
sessionInfo()
```
