# Quality Control


**Loading packages**

```{r, echo=TRUE, results="hide", warning=FALSE, message=FALSE}
library(XMAS2)
library(dplyr)
library(tibble)
library(phyloseq)
library(ggplot2)
```

Quality control of DADA2 results will help us have more rational determinations on the further data analysis. 

## Reads' track by DADA2
```{r, warning=FALSE, message=FALSE, fig.align="center", fig.width=6, fig.height=4, fig.cap="Reads' track by DADA2"}
plot_Dada2Track(data=dada2_res$reads_track)
```


The percentage of the final remained read counts approximate **70%**, indicating that we should consider the sequence depth for analysis when we build the sequence library.

## Spike-in sample (BRS) assessment

The taxonomic levels of spike-in sample's bacteria is genus. Firstly, using the `summarize_taxa` to get the genus level phyloseq object and get the BRS_ID.

```{r, warning=FALSE}
dada2_ps_genus <- summarize_taxa(dada2_ps, taxa_level = "Genus")

dada2_ps_genus@sam_data
```

do `run_RefCheck` under the optimal parameters.

* *BRS_ID*: the ID of BRS sample;

* *Reference*: the directory of the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/);

* *Save*: the directory to save the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/).

To see more details to use `?run_RefCheck`.
```{r, warning=FALSE, collapse=TRUE}
run_RefCheck(
    ps=dada2_ps_genus,
    BRS_ID="S8005",
    Reference=NULL,
    Ref_type="16s",
    Save=NULL)
```


## Spike-in sample's (BRS) remove

After evaluating the sequencing quality, we remove the BRS. 

```{r, warning=FALSE, message=FALSE}
dada2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = dada2_ps,
                     group = "Group",
                     group_names = "QC")
dada2_ps_remove_BRS
```


## Rarefaction curves
```{r, warning=FALSE, message=FALSE, fig.align="center", fig.width=6, fig.height=4, fig.cap="Rarefaction curves"}
plot_RarefCure(ps = dada2_ps_remove_BRS,
               taxa_level = "OTU",
               step = 100,
               label = "Group",
               color = "Group")
```

The result showed that all the samples had **different sequence depth** but had the full sample richness.


## Summarize phyloseq-class object

Summarizing the phyloseq-class object by using `summarize_phyloseq`. It displayed that briefly introduction of the object.

```{r, warning=FALSE, collapse=TRUE}
summarize_phyloseq(dada2_ps_remove_BRS)
```

The minus account of the OTU counts is *51181* in the phyloseq object, and we can use it as the threshold to rarefy. 

Notice the **Sparsity (0.86)**, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.


## Systematic Information
```{r}
sessionInfo()
```