--- 
title: "16s microbiota data analysis based on XMAS package"
author: "Hua Zou"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
always_allow_html: true
description: "16s microbiota data analysis based on XMAS package"
---

# Prerequisites

Here, we give users one example to practice the 16s microbiota data analysis workflow by **XMAS 2.0**. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.


## Preparing for the tutorial

This tutorial requires that the users should have basic knowledge on R language and statistics. If you are new to microbial research, we recommend the **Statistical Analysis of Microbiome Data with R** [@xia2018statistical] as guide book. 

We suggest users using local laptop because our own server have not deployed **XMAS 2.0** package yet. In addition, installing the following software before using **XMAS 2.0**.

* R 3.6.3 or later release [Download link](https://mirrors.tuna.tsinghua.edu.cn/CRAN/).

* Rstudio Desktop [Download link](https://www.rstudio.com/products/rstudio/download/#download).


## Workflow description

The standard data analysis of 16S and MGS data by **XMAS 2.0**. The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:

1. Converting inputs into phyloseq object;

2. Quality Evaluation;

3. Pre-Processing Data;

4. Diversity analysis;

5. Ordination analysis;

6. Composition analysis;

7. Differential analysis.


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.height=8, fig.width=8, fig.align="center", fig.cap="Functions of XMAS 2.0"}
library(Gmisc, quietly = TRUE)
library(glue)
library(htmlTable)
library(grid)
library(magrittr)

ReadsTrack <- boxGrob("Step1: Reads' Track of DADA2\nplot_Dada2Track()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
ImportData <- boxGrob("Step2: Importing data\nimport_dada2_taxa()/get_dada2_phyloseq()/summarize_phyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
RefCheck <- boxGrob("Step3: BRS checking\nsummarize_taxa()/run_RefCheck()/get_GroupPhyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Rarefaction <- boxGrob("Step4: Rarefaction curves\nplot_RarefCurve()/summarize_phyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Rarefy <- boxGrob("Step5: Rarefy OTU table\nnorm_rarefy()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Extract <- boxGrob("Step6: Extract specific taxonomic level\nsummarize_taxa()",
                      txt_gp = gpar(col = "darkblue", cex = .8))

GlobalView <- boxGrob("Step7: GlobalView",
                      txt_gp = gpar(col = "darkblue", cex = .8)) 
GlobalView_a <- boxGrob("Step7.1: Alpha diveristy\nrun_alpha_diversity()/plot_boxplot()",
                      txt_gp = gpar(col = "darkblue", cex = .6)) 
GlobalView_b <- boxGrob("Step7.2: Beta diveristy\nrun_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  
GlobalView_c <- boxGrob("Step7.3: Composition\nplot_stacked_bar_XIVZ()/plot_core_taxa()/core_members()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  

Differential <- boxGrob("Step8: DifferentialAnalysis\nrun_filter()/run_trim()/run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()",
                      txt_gp = gpar(col = "darkblue", cex = .8))  


grid.newpage()
vert <- spreadVertical(ReadsTrack,
                       ImportData = ImportData,
                       RefCheck = RefCheck,
                       Rarefaction = Rarefaction,
                       Rarefy = Rarefy,
                       Extract = Extract,
                       GlobalView = GlobalView,
                       grps = GlobalView_a,
                       Differential = Differential)
grps <- alignVertical(reference = vert$grps,
                      GlobalView_a, GlobalView_b, GlobalView_c) %>%
  spreadHorizontal()

vert$grps <- NULL

for (i in 1:(length(vert) - 1)) {
  connectGrob(vert[[i]], vert[[i + 1]], type = "vert") %>%
    print
}

connectGrob(vert$GlobalView, grps[[1]], type = "N")
connectGrob(vert$GlobalView, grps[[2]], type = "N")
connectGrob(vert$GlobalView, grps[[3]], type = "N")

vert
grps
```

