<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 2 Convert inputdata into phyloseq object | 16s microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="quality-control.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">16s microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2"><i class="fa fa-check"></i><b>2.1</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>2.1.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>2.2</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#reads-track-by-dada2"><i class="fa fa-check"></i><b>3.1</b> Reads’ track by DADA2<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs-assessment"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample (BRS) assessment<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.3</b> Spike-in sample’s (BRS) remove<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="quality-control.html"><a href="quality-control.html#rarefaction-curves"><i class="fa fa-check"></i><b>3.4</b> Rarefaction curves<span></span></a></li>
<li class="chapter" data-level="3.5" data-path="quality-control.html"><a href="quality-control.html#summarize-phyloseq-class-object-1"><i class="fa fa-check"></i><b>3.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#rarefy-otu-counts"><i class="fa fa-check"></i><b>4.1</b> Rarefy otu counts<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.2</b> Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="data-processing.html"><a href="data-processing.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10"><i class="fa fa-check"></i><b>4.3</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="data-processing.html"><a href="data-processing.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>4.4</b> Trimming the taxa with low occurrence less than threshold<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.1</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.2</b> Wilcoxon Rank-Sum test<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.3</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.4</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16s microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="convert-inputdata-into-phyloseq-object" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Convert inputdata into phyloseq object<a href="convert-inputdata-into-phyloseq-object.html#convert-inputdata-into-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><em>r Biocpkg(“phyloseq”)</em> <span class="citation">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> is the most popular <a href="https://bioconductor.org/">Biocondcutor</a> package used by the microbiome research field, and <code>phyloseq-class</code> objects are a great data-standard for microbiota data in R. Therefore, the core functions in <code>XMAS</code> take <code>phyloseq-class</code> object as input. In the phyloseq object, information on OTU abundances, taxonomy of OTUs, the phylogenetic tree and metadata is stored.</p>
<p>This tutorial will introduce you the basic steps to convert results from the in-house pipeline into <em>phyloseq-class object</em>. More importantly on how to look at your data and filter appropriately. We will use the inputs from <strong>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/</strong>.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="convert-inputdata-into-phyloseq-object.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb1-2"><a href="convert-inputdata-into-phyloseq-object.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="convert-inputdata-into-phyloseq-object.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-4"><a href="convert-inputdata-into-phyloseq-object.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-5"><a href="convert-inputdata-into-phyloseq-object.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-6"><a href="convert-inputdata-into-phyloseq-object.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<div id="dada2" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> DADA2<a href="convert-inputdata-into-phyloseq-object.html#dada2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>dada2</strong> results from standardized_analytics_workflow_R_function</p>
<div id="importing-results-from-dada2-pipeline" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Importing results from dada2 pipeline<a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="convert-inputdata-into-phyloseq-object.html#cb2-1" aria-hidden="true" tabindex="-1"></a>dada2_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-2"><a href="convert-inputdata-into-phyloseq-object.html#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb2-3"><a href="convert-inputdata-into-phyloseq-object.html#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_res.rds&quot;</span>,</span>
<span id="cb2-4"><a href="convert-inputdata-into-phyloseq-object.html#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb2-5"><a href="convert-inputdata-into-phyloseq-object.html#cb2-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-6"><a href="convert-inputdata-into-phyloseq-object.html#cb2-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-7"><a href="convert-inputdata-into-phyloseq-object.html#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="convert-inputdata-into-phyloseq-object.html#cb2-8" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb2-9"><a href="convert-inputdata-into-phyloseq-object.html#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb2-10"><a href="convert-inputdata-into-phyloseq-object.html#cb2-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_metadata.tsv&quot;</span>,</span>
<span id="cb2-11"><a href="convert-inputdata-into-phyloseq-object.html#cb2-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb2-12"><a href="convert-inputdata-into-phyloseq-object.html#cb2-12" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-13"><a href="convert-inputdata-into-phyloseq-object.html#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb2-14"><a href="convert-inputdata-into-phyloseq-object.html#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-15"><a href="convert-inputdata-into-phyloseq-object.html#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb2-16"><a href="convert-inputdata-into-phyloseq-object.html#cb2-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-17"><a href="convert-inputdata-into-phyloseq-object.html#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="convert-inputdata-into-phyloseq-object.html#cb2-18" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read_tree</span>(</span>
<span id="cb2-19"><a href="convert-inputdata-into-phyloseq-object.html#cb2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb2-20"><a href="convert-inputdata-into-phyloseq-object.html#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tree.nwk&quot;</span>,</span>
<span id="cb2-21"><a href="convert-inputdata-into-phyloseq-object.html#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb2-22"><a href="convert-inputdata-into-phyloseq-object.html#cb2-22" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-23"><a href="convert-inputdata-into-phyloseq-object.html#cb2-23" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="taxa-table" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> taxa table<a href="convert-inputdata-into-phyloseq-object.html#taxa-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We use <code>import_dada2_taxa</code> to convert dada2_res$tax_tab into our own taxa table</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="convert-inputdata-into-phyloseq-object.html#cb3-1" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">import_dada2_taxa</span>(dada2_res<span class="sc">$</span>tax_tab)</span>
<span id="cb3-2"><a href="convert-inputdata-into-phyloseq-object.html#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="convert-inputdata-into-phyloseq-object.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tax_tab, <span class="dv">1</span>)</span></code></pre></div>
<pre><code>##                                                                                                                                                                                                                                                                      Kingdom
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG k__Bacteria
##                                                                                                                                                                                                                                                                         Phylum
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG p__Firmicutes
##                                                                                                                                                                                                                                                                          Class
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG c__Clostridia
##                                                                                                                                                                                                                                                                             Order
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG o__Clostridiales
##                                                                                                                                                                                                                                                                              Family
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG f__Lachnospiraceae
##                                                                                                                                                                                                                                                                       Genus
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG g__Blautia
##                                                                                                                                                                                                                                                                                  Species
## TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG s__Blautia_unclassified</code></pre>
</div>
<div id="otu-table" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> otu table<a href="convert-inputdata-into-phyloseq-object.html#otu-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="convert-inputdata-into-phyloseq-object.html#cb5-1" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> dada2_res<span class="sc">$</span>seq_tab</span>
<span id="cb5-2"><a href="convert-inputdata-into-phyloseq-object.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb5-3"><a href="convert-inputdata-into-phyloseq-object.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(otu_tab))</span>
<span id="cb5-4"><a href="convert-inputdata-into-phyloseq-object.html#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="convert-inputdata-into-phyloseq-object.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(otu_tab[, <span class="dv">1</span>, F])</span></code></pre></div>
<pre><code>##       TAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGG
## S6030                                                                                                                                                                                                                                                             1253
## S6032                                                                                                                                                                                                                                                            14677
## S6033                                                                                                                                                                                                                                                             2929
## S6035                                                                                                                                                                                                                                                                0
## S6036                                                                                                                                                                                                                                                             3548
## S6037                                                                                                                                                                                                                                                             8484</code></pre>
</div>
<div id="metadata-table" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> metadata table<a href="convert-inputdata-into-phyloseq-object.html#metadata-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="convert-inputdata-into-phyloseq-object.html#cb7-1" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;seqID&quot;</span>)</span>
<span id="cb7-2"><a href="convert-inputdata-into-phyloseq-object.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb7-3"><a href="convert-inputdata-into-phyloseq-object.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(sam_tab))</span>
<span id="cb7-4"><a href="convert-inputdata-into-phyloseq-object.html#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="convert-inputdata-into-phyloseq-object.html#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sam_tab)</span></code></pre></div>
<pre><code>##       Group
## S6065    AA
## S6049    AA
## S6043    AA
## S6037    AA
## S6059    AA
## S6060    AA</code></pre>
</div>
<div id="phyloseq-object" class="section level3 hasAnchor" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> phyloseq object<a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="convert-inputdata-into-phyloseq-object.html#cb9-1" aria-hidden="true" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">import_dada2</span>(<span class="at">seq_tab =</span> otu_tab, </span>
<span id="cb9-2"><a href="convert-inputdata-into-phyloseq-object.html#cb9-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tax_tab =</span> tax_tab, </span>
<span id="cb9-3"><a href="convert-inputdata-into-phyloseq-object.html#cb9-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">sam_tab =</span> sam_tab, </span>
<span id="cb9-4"><a href="convert-inputdata-into-phyloseq-object.html#cb9-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">phy_tree =</span> tree)</span>
<span id="cb9-5"><a href="convert-inputdata-into-phyloseq-object.html#cb9-5" aria-hidden="true" tabindex="-1"></a>dada2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 24 samples ]
## sample_data() Sample Data:       [ 24 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
<p>we obtain the <strong>phyloseq-class object</strong> and then use it to perform data analysis.</p>
<p>Here, the phyloseq object comprises five components (<em>OTU Table</em>, <em>Sample Data</em>, <em>Taxonomy Table</em>, <em>Phylogenetic Tree</em> and <em>DNAStringSet</em>).</p>
<ul>
<li>otu_table</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="convert-inputdata-into-phyloseq-object.html#cb11-1" aria-hidden="true" tabindex="-1"></a>dada2_ps<span class="sc">@</span>otu_table <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##       S6030 S6032 S6033 S6035 S6036 S6037 S6040 S6043 S6045 S6046 S6048 S6049 S6050 S6054 S6055 S6058 S6059 S6060 S6061 S6063 S6065
## ASV_1  1253 14677  2929     0  3548  8484 11727  5218  4942  7347  3503  3803 14593  4275  2812     0  6847 17893  2956  3328  6302
## ASV_2  2810  1558  1839 11885     0  5632   631  7326  1809  4781  3476   108  3100    39  3879  6389   298  4995   196 45061   542
## ASV_3  7107  2915  2200     0  1232 19675     0  1617   839  9362  4815     0     0  1121 13472     0  1421     0     0     0  9694
## ASV_4     0  2280    94 13775     0  4456   937   746  3730   273    85    64   562   448    18   611    96  5976     0   837     0
## ASV_5  6983 15963  1172   256 12140   874  1722   120   210     0   111   802   994  1971    18   517   808   443    48   109     7
## ASV_6   489     0   199     0   836     0  8565     0     0  1257     0  5998     0  3093  2091     0 10572     0  8495     0  2703
##       S6066 S6068 S8005
## ASV_1  8414     0     0
## ASV_2   357 10870  7325
## ASV_3  4387     0 14797
## ASV_4   706 17977     0
## ASV_5   122   546  2516
## ASV_6     0     0   974</code></pre>
<ul>
<li>tax_table</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="convert-inputdata-into-phyloseq-object.html#cb13-1" aria-hidden="true" tabindex="-1"></a>dada2_ps<span class="sc">@</span>tax_table <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##           Kingdom            Phylum             Class                Order                Family                           Genus
## ASV_1 k__Bacteria     p__Firmicutes     c__Clostridia     o__Clostridiales    f__Lachnospiraceae                      g__Blautia
## ASV_2 k__Bacteria p__Actinobacteria c__Actinobacteria o__Bifidobacteriales f__Bifidobacteriaceae              g__Bifidobacterium
## ASV_3 k__Bacteria p__Actinobacteria c__Actinobacteria o__Bifidobacteriales f__Bifidobacteriaceae              g__Bifidobacterium
## ASV_4 k__Bacteria     p__Firmicutes     c__Clostridia     o__Clostridiales    f__Lachnospiraceae g__Lachnospiraceae_unclassified
## ASV_5 k__Bacteria     p__Firmicutes        c__Bacilli   o__Lactobacillales   f__Streptococcaceae                g__Streptococcus
## ASV_6 k__Bacteria     p__Firmicutes     c__Clostridia     o__Clostridiales    f__Ruminococcaceae             g__Faecalibacterium
##                                Species
## ASV_1          s__Blautia_unclassified
## ASV_2  s__Bifidobacterium_unclassified
## ASV_3  s__Bifidobacterium_unclassified
## ASV_4  s__Lachnospiraceae_unclassified
## ASV_5    s__Streptococcus_unclassified
## ASV_6 s__Faecalibacterium_unclassified</code></pre>
<ul>
<li>sample_table</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="convert-inputdata-into-phyloseq-object.html#cb15-1" aria-hidden="true" tabindex="-1"></a>dada2_ps<span class="sc">@</span>sam_data <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##       Group
## S6030    BB
## S6032    BB
## S6033    BB
## S6035    AA
## S6036    BB
## S6037    AA</code></pre>
</div>
</div>
<div id="summarize-phyloseq-class-object" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Summarize phyloseq-class object<a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="convert-inputdata-into-phyloseq-object.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps)</span>
<span id="cb17-2"><a href="convert-inputdata-into-phyloseq-object.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb17-3"><a href="convert-inputdata-into-phyloseq-object.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 936223] Total number of reads = 15025374] Average number of reads = 62605.70833333335] Median number of reads = 619157] Sparsity = 0.8653738839285716] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons</span></span>
<span id="cb17-4"><a href="convert-inputdata-into-phyloseq-object.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb17-5"><a href="convert-inputdata-into-phyloseq-object.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb17-6"><a href="convert-inputdata-into-phyloseq-object.html#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb17-7"><a href="convert-inputdata-into-phyloseq-object.html#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-8"><a href="convert-inputdata-into-phyloseq-object.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb17-9"><a href="convert-inputdata-into-phyloseq-object.html#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 93622&quot;</span></span>
<span id="cb17-10"><a href="convert-inputdata-into-phyloseq-object.html#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-11"><a href="convert-inputdata-into-phyloseq-object.html#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb17-12"><a href="convert-inputdata-into-phyloseq-object.html#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1502537&quot;</span></span>
<span id="cb17-13"><a href="convert-inputdata-into-phyloseq-object.html#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-14"><a href="convert-inputdata-into-phyloseq-object.html#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb17-15"><a href="convert-inputdata-into-phyloseq-object.html#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 62605.7083333333&quot;</span></span>
<span id="cb17-16"><a href="convert-inputdata-into-phyloseq-object.html#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-17"><a href="convert-inputdata-into-phyloseq-object.html#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb17-18"><a href="convert-inputdata-into-phyloseq-object.html#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61915&quot;</span></span>
<span id="cb17-19"><a href="convert-inputdata-into-phyloseq-object.html#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-20"><a href="convert-inputdata-into-phyloseq-object.html#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb17-21"><a href="convert-inputdata-into-phyloseq-object.html#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.865373883928571&quot;</span></span>
<span id="cb17-22"><a href="convert-inputdata-into-phyloseq-object.html#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-23"><a href="convert-inputdata-into-phyloseq-object.html#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb17-24"><a href="convert-inputdata-into-phyloseq-object.html#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;</span></span>
<span id="cb17-25"><a href="convert-inputdata-into-phyloseq-object.html#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-26"><a href="convert-inputdata-into-phyloseq-object.html#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb17-27"><a href="convert-inputdata-into-phyloseq-object.html#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 0&quot;</span></span>
<span id="cb17-28"><a href="convert-inputdata-into-phyloseq-object.html#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-29"><a href="convert-inputdata-into-phyloseq-object.html#cb17-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb17-30"><a href="convert-inputdata-into-phyloseq-object.html#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb17-31"><a href="convert-inputdata-into-phyloseq-object.html#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-32"><a href="convert-inputdata-into-phyloseq-object.html#cb17-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb17-33"><a href="convert-inputdata-into-phyloseq-object.html#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb17-34"><a href="convert-inputdata-into-phyloseq-object.html#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb17-35"><a href="convert-inputdata-into-phyloseq-object.html#cb17-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb17-36"><a href="convert-inputdata-into-phyloseq-object.html#cb17-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Systematic Information<a href="convert-inputdata-into-phyloseq-object.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="convert-inputdata-into-phyloseq-object.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggpubr_0.4.0    ggplot2_3.3.5   phyloseq_1.38.0 tibble_3.1.6    dplyr_1.0.8     XMAS_2.1.1     
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.4.1             Hmisc_4.6-0                 plyr_1.8.6                  igraph_1.2.11              
##   [5] splines_4.1.2               BiocParallel_1.28.3         GenomeInfoDb_1.30.1         TH.data_1.1-0              
##   [9] lpsymphony_1.22.0           digest_0.6.29               foreach_1.5.2               htmltools_0.5.2            
##  [13] fansi_1.0.2                 checkmate_2.0.0             magrittr_2.0.2              memoise_2.0.1              
##  [17] cluster_2.1.2               limma_3.50.1                Biostrings_2.62.0           annotate_1.72.0            
##  [21] matrixStats_0.61.0          sandwich_3.0-1              metagenomeSeq_1.36.0        jpeg_0.1-9                 
##  [25] colorspace_2.0-3            ggrepel_0.9.1               blob_1.2.2                  corncob_0.2.0              
##  [29] xfun_0.30                   crayon_1.5.0                RCurl_1.98-1.6              jsonlite_1.8.0             
##  [33] libcoin_1.0-9               genefilter_1.76.0           biglm_0.9-2.1               survival_3.3-1             
##  [37] zoo_1.8-9                   iterators_1.0.14            ape_5.6-2                   glue_1.6.2                 
##  [41] gtable_0.3.0                zlibbioc_1.40.0             XVector_0.34.0              DelayedArray_0.20.0        
##  [45] car_3.0-12                  Rhdf5lib_1.16.0             shape_1.4.6                 BiocGenerics_0.40.0        
##  [49] DEoptimR_1.0-10             abind_1.4-5                 scales_1.1.1                mvtnorm_1.1-3              
##  [53] DBI_1.1.2                   edgeR_3.36.0                rstatix_0.7.0               Rcpp_1.0.8.2               
##  [57] htmlTable_2.4.0             xtable_1.8-4                foreign_0.8-82              bit_4.0.4                  
##  [61] Formula_1.2-4               stats4_4.1.2                glmnet_4.1-3                htmlwidgets_1.5.4          
##  [65] httr_1.4.2                  getopt_1.20.3               DiagrammeR_1.0.9            gplots_3.1.1               
##  [69] RColorBrewer_1.1-2          modeltools_0.2-23           ellipsis_0.3.2              pkgconfig_2.0.3            
##  [73] XML_3.99-0.9                nnet_7.3-17                 sass_0.4.0                  locfit_1.5-9.5             
##  [77] utf8_1.2.2                  tidyselect_1.1.2            rlang_1.0.2                 reshape2_1.4.4             
##  [81] AnnotationDbi_1.56.2        munsell_0.5.0               tools_4.1.2                 visNetwork_2.1.0           
##  [85] cachem_1.0.6                cli_3.2.0                   generics_0.1.2              RSQLite_2.2.10             
##  [89] ade4_1.7-18                 broom_0.7.12                evaluate_0.15               biomformat_1.22.0          
##  [93] stringr_1.4.0               fastmap_1.1.0               yaml_2.3.5                  knitr_1.37                 
##  [97] bit64_4.0.5                 robustbase_0.93-9           caTools_1.18.2              purrr_0.3.4                
## [101] KEGGREST_1.34.0             coin_1.4-2                  nlme_3.1-155                protoclust_1.6.3           
## [105] RAIDA_1.0                   compiler_4.1.2              rstudioapi_0.13             png_0.1-7                  
## [109] ggsignif_0.6.3              geneplotter_1.72.0          bslib_0.3.1                 pcaPP_1.9-74               
## [113] stringi_1.7.6               Maaslin2_1.8.0              lattice_0.20-45             Matrix_1.4-0               
## [117] vegan_2.5-7                 permute_0.9-7               multtest_2.50.0             vctrs_0.3.8                
## [121] pillar_1.7.0                lifecycle_1.0.1             rhdf5filters_1.6.0          optparse_1.7.1             
## [125] jquerylib_0.1.4             data.table_1.14.2           cowplot_1.1.1               bitops_1.0-7               
## [129] qvalue_2.26.0               GenomicRanges_1.46.1        latticeExtra_0.6-29         R6_2.5.1                   
## [133] bookdown_0.24               gridExtra_2.3               KernSmooth_2.23-20          IRanges_2.28.0             
## [137] codetools_0.2-18            gtools_3.9.2                MASS_7.3-55                 assertthat_0.2.1           
## [141] Wrench_1.12.0               rhdf5_2.38.1                SummarizedExperiment_1.24.0 DESeq2_1.34.0              
## [145] withr_2.5.0                 multcomp_1.4-18             S4Vectors_0.32.3            GenomeInfoDbData_1.2.7     
## [149] mgcv_1.8-39                 parallel_4.1.2              rpart_4.1.16                grid_4.1.2                 
## [153] tidyr_1.2.0                 rmarkdown_2.13              MatrixGenerics_1.6.0        carData_3.0-5              
## [157] base64enc_0.1-3             Biobase_2.54.0</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-mcmurdie2013phyloseq" class="csl-entry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>“Phyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PloS One</em> 8 (4): e61217.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/16s_DataAnalysis_template/blob/main/01-load-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["16s microbiota data analysis based on XMAS package.pdf", "16s microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
