<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 5 Diversity analysis | 16s microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-processing.html"/>
<link rel="next" href="permanova-beta-dispersion-ordination.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">16s microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2"><i class="fa fa-check"></i><b>2.1</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>2.1.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>2.2</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#reads-track-by-dada2"><i class="fa fa-check"></i><b>3.1</b> Reads’ track by DADA2<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs-assessment"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample (BRS) assessment<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.3</b> Spike-in sample’s (BRS) remove<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="quality-control.html"><a href="quality-control.html#rarefaction-curves"><i class="fa fa-check"></i><b>3.4</b> Rarefaction curves<span></span></a></li>
<li class="chapter" data-level="3.5" data-path="quality-control.html"><a href="quality-control.html#summarize-phyloseq-class-object-1"><i class="fa fa-check"></i><b>3.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#rarefy-otu-counts"><i class="fa fa-check"></i><b>4.1</b> Rarefy otu counts<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.2</b> Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-and-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering and trimming<span></span></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10"><i class="fa fa-check"></i><b>8.1.1</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)<span></span></a></li>
<li class="chapter" data-level="8.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.1.2</b> Trimming the taxa with low occurrence less than threshold<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.3</b> Wilcoxon Rank-Sum test<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="test-example.html"><a href="test-example.html"><i class="fa fa-check"></i><b>9</b> Test Example<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="test-example.html"><a href="test-example.html#loading-packages"><i class="fa fa-check"></i><b>9.1</b> Loading packages<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="test-example.html"><a href="test-example.html#loading-data"><i class="fa fa-check"></i><b>9.2</b> Loading data<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="test-example.html"><a href="test-example.html#metaphlan2-result"><i class="fa fa-check"></i><b>9.3</b> Metaphlan2 result<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="test-example.html"><a href="test-example.html#step1-reads-track"><i class="fa fa-check"></i><b>9.4</b> Step1: Reads’ Track<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="test-example.html"><a href="test-example.html#step2-convert-inputs-into-phyloseq-data"><i class="fa fa-check"></i><b>9.5</b> Step2: Convert inputs into phyloseq data<span></span></a></li>
<li class="chapter" data-level="9.6" data-path="test-example.html"><a href="test-example.html#step3-brs-checking"><i class="fa fa-check"></i><b>9.6</b> Step3: BRS checking<span></span></a></li>
<li class="chapter" data-level="9.7" data-path="test-example.html"><a href="test-example.html#step4-rarefaction-curves"><i class="fa fa-check"></i><b>9.7</b> Step4: Rarefaction curves<span></span></a></li>
<li class="chapter" data-level="9.8" data-path="test-example.html"><a href="test-example.html#step5-rarefy-otu-counts"><i class="fa fa-check"></i><b>9.8</b> Step5: Rarefy otu counts<span></span></a></li>
<li class="chapter" data-level="9.9" data-path="test-example.html"><a href="test-example.html#step6-extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>9.9</b> Step6: Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="9.10" data-path="test-example.html"><a href="test-example.html#step7-globalview"><i class="fa fa-check"></i><b>9.10</b> Step7: GlobalView<span></span></a></li>
<li class="chapter" data-level="9.11" data-path="test-example.html"><a href="test-example.html#step8-differential-analysis"><i class="fa fa-check"></i><b>9.11</b> Step8: Differential Analysis<span></span></a></li>
<li class="chapter" data-level="9.12" data-path="test-example.html"><a href="test-example.html#systematic-information-7"><i class="fa fa-check"></i><b>9.12</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16s microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="diversity-analysis" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Diversity analysis<a href="diversity-analysis.html#diversity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="diversity-analysis.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb39-2"><a href="diversity-analysis.html#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb39-3"><a href="diversity-analysis.html#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb39-4"><a href="diversity-analysis.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb39-5"><a href="diversity-analysis.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="alpha-diveristy" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Alpha diveristy<a href="diversity-analysis.html#alpha-diveristy" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> the otu table must be counts matrix (<em>rarefied but not trim counts matrix</em>) when you choose <em>Observed</em> etc measures.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="diversity-analysis.html#cb40-1" aria-hidden="true" tabindex="-1"></a>dada_ps_rare_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>dada2_ps_rare_genus, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb40-2"><a href="diversity-analysis.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada_ps_rare_genus_alpha)</span></code></pre></div>
<pre><code>##    SampleID Group Observed    Chao1  se.chao1  Shannon
## 1     S6030    BB       74 74.00000 0.0000000 2.622800
## 2     S6032    BB       38 38.00000 0.0000000 1.711068
## 3     S6033    BB       80 80.16667 0.5431879 2.719495
## 4     S6035    AA       35 35.00000 0.1232013 1.953807
## 5     S6036    BB       95 95.00000 0.0000000 2.942657
## 6     S6037    AA       35 35.00000 0.0000000 1.933007
## 7     S6040    BB       58 58.00000 0.0000000 2.490198
## 8     S6043    AA       65 65.33333 0.9246628 2.098003
## 9     S6045    BB       60 60.00000 0.0000000 2.920363
## 10    S6046    BB       71 71.00000 0.1241166 2.264433
## 11    S6048    BB       66 66.00000 0.0000000 2.630421
## 12    S6049    AA       60 60.00000 0.0000000 2.506885
## 13    S6050    BB       53 53.00000 0.0000000 2.431334
## 14    S6054    BB       84 84.00000 0.0000000 2.769973
## 15    S6055    BB       75 75.00000 0.0000000 2.297388
## 16    S6058    BB       37 37.00000 0.2465985 2.262059
## 17    S6059    AA       71 71.00000 0.0000000 2.682960
## 18    S6060    AA       48 48.00000 0.0000000 1.883809
## 19    S6061    AA       80 80.00000 0.0000000 3.088651
## 20    S6063    BB       38 38.00000 0.0000000 1.543267
## 21    S6065    AA       77 77.00000 0.0000000 2.877568
## 22    S6066    AA       67 67.00000 0.0000000 2.638564
## 23    S6068    BB       40 40.00000 0.0000000 2.100363</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="diversity-analysis.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data=</span>dada_ps_rare_genus_alpha,</span>
<span id="cb42-2"><a href="diversity-analysis.html#cb42-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb42-3"><a href="diversity-analysis.html#cb42-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb42-4"><a href="diversity-analysis.html#cb42-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb42-5"><a href="diversity-analysis.html#cb42-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-37"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-37-1.png" alt="Alpha diversity (16s example)" width="576" />
<p class="caption">
Figure 5.1: Alpha diversity (16s example)
</p>
</div>
</div>
<div id="beta-diversity" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Beta diversity<a href="diversity-analysis.html#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="diversity-analysis.html#cb43-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rare_genus, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.017526 0.0175257 2.7119    999  0.109
## Residuals 21 0.135711 0.0064624                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA    BB
## AA         0.116
## BB 0.11449</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="diversity-analysis.html#cb45-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta<span class="sc">$</span>BetaDispersion</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-38"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-38-1.png" alt="Beta diversity (16s example)" width="480" />
<p class="caption">
Figure 5.2: Beta diversity (16s example)
</p>
</div>
<ul>
<li>beta distance</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="diversity-analysis.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(dada2_ps_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-39"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-39-1.png" alt="Beta distance" width="480" />
<p class="caption">
Figure 5.3: Beta distance
</p>
</div>
</div>
<div id="systematic-information-3" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Systematic Information<a href="diversity-analysis.html#systematic-information-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="diversity-analysis.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] magrittr_2.0.2              glue_1.6.2                  Gmisc_3.0.0                 htmlTable_2.4.0            
##  [5] Rcpp_1.0.8.2                readxl_1.4.0                SummarizedExperiment_1.24.0 Biobase_2.54.0             
##  [9] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.3           
## [13] BiocGenerics_0.40.0         MatrixGenerics_1.6.0        matrixStats_0.61.0          ggpubr_0.4.0               
## [17] ggplot2_3.3.5               phyloseq_1.38.0             tibble_3.1.6                dplyr_1.0.8                
## [21] XMAS2_2.1.3                
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.4.1        Hmisc_4.6-0            corrplot_0.92          plyr_1.8.6             igraph_1.2.11         
##   [6] splines_4.1.2          BiocParallel_1.28.3    TH.data_1.1-0          digest_0.6.29          foreach_1.5.2         
##  [11] htmltools_0.5.2        fansi_1.0.2            checkmate_2.0.0        memoise_2.0.1          cluster_2.1.2         
##  [16] limma_3.50.1           Biostrings_2.62.0      annotate_1.72.0        sandwich_3.0-1         metagenomeSeq_1.36.0  
##  [21] jpeg_0.1-9             colorspace_2.0-3       blob_1.2.2             ggrepel_0.9.1          xfun_0.30             
##  [26] crayon_1.5.0           RCurl_1.98-1.6         jsonlite_1.8.0         libcoin_1.0-9          genefilter_1.76.0     
##  [31] survival_3.3-1         zoo_1.8-9              iterators_1.0.14       ape_5.6-2              gtable_0.3.0          
##  [36] zlibbioc_1.40.0        XVector_0.34.0         DelayedArray_0.20.0    car_3.0-12             Rhdf5lib_1.16.0       
##  [41] shape_1.4.6            abind_1.4-5            scales_1.1.1           mvtnorm_1.1-3          DBI_1.1.2             
##  [46] edgeR_3.36.0           rstatix_0.7.0          xtable_1.8-4           foreign_0.8-82         bit_4.0.4             
##  [51] Formula_1.2-4          glmnet_4.1-3           htmlwidgets_1.5.4      httr_1.4.2             gplots_3.1.1          
##  [56] RColorBrewer_1.1-2     modeltools_0.2-23      ellipsis_0.3.2         pkgconfig_2.0.3        XML_3.99-0.9          
##  [61] farver_2.1.0           nnet_7.3-17            sass_0.4.0             locfit_1.5-9.5         utf8_1.2.2            
##  [66] tidyselect_1.1.2       labeling_0.4.2         rlang_1.0.2            reshape2_1.4.4         AnnotationDbi_1.56.2  
##  [71] munsell_0.5.0          cellranger_1.1.0       tools_4.1.2            cachem_1.0.6           cli_3.2.0             
##  [76] generics_0.1.2         RSQLite_2.2.10         ade4_1.7-18            broom_0.7.12           evaluate_0.15         
##  [81] biomformat_1.22.0      stringr_1.4.0          fastmap_1.1.0          yaml_2.3.5             knitr_1.37            
##  [86] bit64_4.0.5            caTools_1.18.2         forestplot_2.0.1       purrr_0.3.4            KEGGREST_1.34.0       
##  [91] coin_1.4-2             nlme_3.1-155           compiler_4.1.2         rstudioapi_0.13        png_0.1-7             
##  [96] ggsignif_0.6.3         geneplotter_1.72.0     bslib_0.3.1            stringi_1.7.6          highr_0.9             
## [101] lattice_0.20-45        Matrix_1.4-0           vegan_2.5-7            permute_0.9-7          multtest_2.50.0       
## [106] vctrs_0.3.8            pillar_1.7.0           lifecycle_1.0.1        rhdf5filters_1.6.0     jquerylib_0.1.4       
## [111] data.table_1.14.2      cowplot_1.1.1          bitops_1.0-7           latticeExtra_0.6-29    R6_2.5.1              
## [116] bookdown_0.24          gridExtra_2.3          KernSmooth_2.23-20     codetools_0.2-18       MASS_7.3-55           
## [121] gtools_3.9.2           assertthat_0.2.1       Wrench_1.12.0          rhdf5_2.38.1           DESeq2_1.34.0         
## [126] withr_2.5.0            multcomp_1.4-18        GenomeInfoDbData_1.2.7 mgcv_1.8-39            parallel_4.1.2        
## [131] rpart_4.1.16           tidyr_1.2.0            rmarkdown_2.13         carData_3.0-5          lubridate_1.8.0       
## [136] base64enc_0.1-3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="permanova-beta-dispersion-ordination.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/16s_DataAnalysis_template/blob/main/04-Diversity.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["16s microbiota data analysis based on XMAS package.pdf", "16s microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
