<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 9 Test Example | 16s microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-analysis.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">16s microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2"><i class="fa fa-check"></i><b>2.1</b> DADA2</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>2.1.1</b> Importing results from dada2 pipeline</a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table</a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table</a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table</a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>2.2</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#reads-track-by-dada2"><i class="fa fa-check"></i><b>3.1</b> Reads’ track by DADA2</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs-assessment"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample (BRS) assessment</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.3</b> Spike-in sample’s (BRS) remove</a></li>
<li class="chapter" data-level="3.4" data-path="quality-control.html"><a href="quality-control.html#rarefaction-curves"><i class="fa fa-check"></i><b>3.4</b> Rarefaction curves</a></li>
<li class="chapter" data-level="3.5" data-path="quality-control.html"><a href="quality-control.html#summarize-phyloseq-class-object-1"><i class="fa fa-check"></i><b>3.5</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="3.6" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#rarefy-otu-counts"><i class="fa fa-check"></i><b>4.1</b> Rarefy otu counts</a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.2</b> Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="4.3" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy</a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity</a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination</a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion</a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination</a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition</a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot</a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota</a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-and-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering and trimming</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10"><i class="fa fa-check"></i><b>8.1.1</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)</a></li>
<li class="chapter" data-level="8.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.1.2</b> Trimming the taxa with low occurrence less than threshold</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.3</b> Wilcoxon Rank-Sum test</a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="test-example.html"><a href="test-example.html"><i class="fa fa-check"></i><b>9</b> Test Example</a>
<ul>
<li class="chapter" data-level="9.1" data-path="test-example.html"><a href="test-example.html#loading-packages"><i class="fa fa-check"></i><b>9.1</b> Loading packages</a></li>
<li class="chapter" data-level="9.2" data-path="test-example.html"><a href="test-example.html#loading-data"><i class="fa fa-check"></i><b>9.2</b> Loading data</a></li>
<li class="chapter" data-level="9.3" data-path="test-example.html"><a href="test-example.html#metaphlan2-result"><i class="fa fa-check"></i><b>9.3</b> Metaphlan2 result</a></li>
<li class="chapter" data-level="9.4" data-path="test-example.html"><a href="test-example.html#step1-reads-track"><i class="fa fa-check"></i><b>9.4</b> Step1: Reads’ Track</a></li>
<li class="chapter" data-level="9.5" data-path="test-example.html"><a href="test-example.html#step2-convert-inputs-into-phyloseq-data"><i class="fa fa-check"></i><b>9.5</b> Step2: Convert inputs into phyloseq data</a></li>
<li class="chapter" data-level="9.6" data-path="test-example.html"><a href="test-example.html#step3-brs-checking"><i class="fa fa-check"></i><b>9.6</b> Step3: BRS checking</a></li>
<li class="chapter" data-level="9.7" data-path="test-example.html"><a href="test-example.html#step4-rarefaction-curves"><i class="fa fa-check"></i><b>9.7</b> Step4: Rarefaction curves</a></li>
<li class="chapter" data-level="9.8" data-path="test-example.html"><a href="test-example.html#step5-rarefy-otu-counts"><i class="fa fa-check"></i><b>9.8</b> Step5: Rarefy otu counts</a></li>
<li class="chapter" data-level="9.9" data-path="test-example.html"><a href="test-example.html#step6-extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>9.9</b> Step6: Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="9.10" data-path="test-example.html"><a href="test-example.html#step7-globalview"><i class="fa fa-check"></i><b>9.10</b> Step7: GlobalView</a></li>
<li class="chapter" data-level="9.11" data-path="test-example.html"><a href="test-example.html#step8-differential-analysis"><i class="fa fa-check"></i><b>9.11</b> Step8: Differential Analysis</a></li>
<li class="chapter" data-level="9.12" data-path="test-example.html"><a href="test-example.html#systematic-information-7"><i class="fa fa-check"></i><b>9.12</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16s microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="test-example" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Test Example<a href="test-example.html#test-example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Loading packages<a href="test-example.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="test-example.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb87-2"><a href="test-example.html#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb87-3"><a href="test-example.html#cb87-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb87-4"><a href="test-example.html#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb87-5"><a href="test-example.html#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb87-6"><a href="test-example.html#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb87-7"><a href="test-example.html#cb87-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code></pre></div>
</div>
<div id="loading-data" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Loading data<a href="test-example.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="test-example.html#cb88-1" aria-hidden="true" tabindex="-1"></a>dada2_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/RawData/dada2_res.rds&quot;</span>)</span>
<span id="cb88-2"><a href="test-example.html#cb88-2" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">read_tree</span>(<span class="st">&quot;DataSet/RawData/tree.nwk&quot;</span>)</span>
<span id="cb88-3"><a href="test-example.html#cb88-3" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;DataSet/RawData/诺禾宏基因组678月-ZH.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">3</span>)</span>
<span id="cb88-4"><a href="test-example.html#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="test-example.html#cb88-5" aria-hidden="true" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;DataSet/RawData/merged_metaphlan2.tsv&quot;</span>,</span>
<span id="cb88-6"><a href="test-example.html#cb88-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-7"><a href="test-example.html#cb88-7" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span></code></pre></div>
</div>
<div id="metaphlan2-result" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Metaphlan2 result<a href="test-example.html#metaphlan2-result" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="test-example.html#cb89-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(<span class="at">data_metaphlan2 =</span> metaphlan2_res, </span>
<span id="cb89-2"><a href="test-example.html#cb89-2" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb89-3"><a href="test-example.html#cb89-3" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb89-4"><a href="test-example.html#cb89-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-5"><a href="test-example.html#cb89-5" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb89-6"><a href="test-example.html#cb89-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;S_&quot;</span>, <span class="fu">colnames</span>(otu_tab))</span>
<span id="cb89-7"><a href="test-example.html#cb89-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-8"><a href="test-example.html#cb89-8" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-9"><a href="test-example.html#cb89-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;粪便&quot;</span>, <span class="st">&quot;Stool&quot;</span>, </span>
<span id="cb89-10"><a href="test-example.html#cb89-10" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;Product&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb89-11"><a href="test-example.html#cb89-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SampleTubeID, Group, <span class="fu">everything</span>())</span>
<span id="cb89-12"><a href="test-example.html#cb89-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S_&quot;</span>, sam_tab<span class="sc">$</span>SeqID_MGS)</span>
<span id="cb89-13"><a href="test-example.html#cb89-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-14"><a href="test-example.html#cb89-14" aria-hidden="true" tabindex="-1"></a>overlap_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(sam_tab), <span class="fu">colnames</span>(otu_tab))</span>
<span id="cb89-15"><a href="test-example.html#cb89-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-16"><a href="test-example.html#cb89-16" aria-hidden="true" tabindex="-1"></a>otu_tab_cln <span class="ot">&lt;-</span> otu_tab[, <span class="fu">match</span>(overlap_samples, <span class="fu">colnames</span>(otu_tab))]</span>
<span id="cb89-17"><a href="test-example.html#cb89-17" aria-hidden="true" tabindex="-1"></a>sam_tab_cln <span class="ot">&lt;-</span> sam_tab[<span class="fu">match</span>(overlap_samples, <span class="fu">rownames</span>(sam_tab)), ]</span>
<span id="cb89-18"><a href="test-example.html#cb89-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab_cln) <span class="ot">&lt;-</span> overlap_samples</span>
<span id="cb89-19"><a href="test-example.html#cb89-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-20"><a href="test-example.html#cb89-20" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">get_metaphlan_phyloseq</span>(</span>
<span id="cb89-21"><a href="test-example.html#cb89-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">otu_tab =</span> otu_tab_cln, </span>
<span id="cb89-22"><a href="test-example.html#cb89-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sam_tab =</span> sam_tab_cln,</span>
<span id="cb89-23"><a href="test-example.html#cb89-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb89-24"><a href="test-example.html#cb89-24" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 328 taxa and 145 samples ]
## sample_data() Sample Data:       [ 145 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 328 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="step1-reads-track" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Step1: Reads’ Track<a href="test-example.html#step1-reads-track" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="test-example.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(<span class="at">data =</span> dada2_res<span class="sc">$</span>reads_track) <span class="sc">+</span></span>
<span id="cb91-2"><a href="test-example.html#cb91-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-57"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-57-1.png" alt="DADA2' read track (Example)" width="768" />
<p class="caption">
Figure 9.1: DADA2’ read track (Example)
</p>
</div>
</div>
<div id="step2-convert-inputs-into-phyloseq-data" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Step2: Convert inputs into phyloseq data<a href="test-example.html#step2-convert-inputs-into-phyloseq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="test-example.html#cb92-1" aria-hidden="true" tabindex="-1"></a>tax_tab_16s <span class="ot">&lt;-</span> <span class="fu">import_dada2_taxa</span>(<span class="at">dada2_taxa =</span> dada2_res<span class="sc">$</span>tax_tab)</span>
<span id="cb92-2"><a href="test-example.html#cb92-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-3"><a href="test-example.html#cb92-3" aria-hidden="true" tabindex="-1"></a>otu_tab_16s <span class="ot">&lt;-</span> dada2_res<span class="sc">$</span>seq_tab</span>
<span id="cb92-4"><a href="test-example.html#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb92-5"><a href="test-example.html#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(otu_tab_16s) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S_&quot;</span>, <span class="fu">rownames</span>(otu_tab_16s))</span>
<span id="cb92-6"><a href="test-example.html#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="test-example.html#cb92-7" aria-hidden="true" tabindex="-1"></a>sam_tab_16s <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb92-8"><a href="test-example.html#cb92-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;粪便&quot;</span>, <span class="st">&quot;Stool&quot;</span>, </span>
<span id="cb92-9"><a href="test-example.html#cb92-9" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;Product&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb92-10"><a href="test-example.html#cb92-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleTubeID <span class="sc">%in%</span> sam_tab_cln<span class="sc">$</span>SampleTubeID) <span class="sc">%&gt;%</span> </span>
<span id="cb92-11"><a href="test-example.html#cb92-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SampleTubeID, Group, <span class="fu">everything</span>())</span>
<span id="cb92-12"><a href="test-example.html#cb92-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab_16s) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S_&quot;</span>, sam_tab_16s<span class="sc">$</span>SeqID_16s)</span>
<span id="cb92-13"><a href="test-example.html#cb92-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-14"><a href="test-example.html#cb92-14" aria-hidden="true" tabindex="-1"></a>overlap_samples_16s <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(sam_tab_16s), <span class="fu">rownames</span>(otu_tab_16s))</span>
<span id="cb92-15"><a href="test-example.html#cb92-15" aria-hidden="true" tabindex="-1"></a>otu_tab_16s_cln <span class="ot">&lt;-</span> otu_tab_16s[<span class="fu">match</span>(overlap_samples_16s, <span class="fu">rownames</span>(otu_tab_16s)), ]</span>
<span id="cb92-16"><a href="test-example.html#cb92-16" aria-hidden="true" tabindex="-1"></a>sam_tab_16s_cln <span class="ot">&lt;-</span> sam_tab_16s[<span class="fu">match</span>(overlap_samples_16s, <span class="fu">rownames</span>(sam_tab_16s)), ]</span>
<span id="cb92-17"><a href="test-example.html#cb92-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-18"><a href="test-example.html#cb92-18" aria-hidden="true" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">get_dada2_phyloseq</span>(</span>
<span id="cb92-19"><a href="test-example.html#cb92-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">seq_tab =</span> otu_tab_16s_cln, </span>
<span id="cb92-20"><a href="test-example.html#cb92-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">tax_tab =</span> tax_tab_16s, </span>
<span id="cb92-21"><a href="test-example.html#cb92-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">sam_tab =</span> sam_tab_16s_cln, </span>
<span id="cb92-22"><a href="test-example.html#cb92-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">phy_tree =</span> tree)</span>
<span id="cb92-23"><a href="test-example.html#cb92-23" aria-hidden="true" tabindex="-1"></a>dada2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1948 taxa and 145 samples ]
## sample_data() Sample Data:       [ 145 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 1948 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1948 tips and 1933 internal nodes ]
## refseq()      DNAStringSet:      [ 1948 reference sequences ]</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="test-example.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step2/&quot;</span>)) {</span>
<span id="cb94-2"><a href="test-example.html#cb94-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step2/&quot;</span>)</span>
<span id="cb94-3"><a href="test-example.html#cb94-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb94-4"><a href="test-example.html#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps, <span class="st">&quot;DataSet/Step2/Donor_16s_phyloseq.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step3-brs-checking" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Step3: BRS checking<a href="test-example.html#step3-brs-checking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="test-example.html#cb95-1" aria-hidden="true" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step2/Donor_16s_phyloseq.RDS&quot;</span>)</span>
<span id="cb95-2"><a href="test-example.html#cb95-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps, </span>
<span id="cb95-3"><a href="test-example.html#cb95-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb95-4"><a href="test-example.html#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(dada2_ps_genus<span class="sc">@</span>sam_data <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>())</span></code></pre></div>
<pre><code>##             SampleTubeID   Group Date_Sequencing ProductID SampleType ProductBatch Date_Sampling Date_Receiving SeqID_MGS
## S_7929      GGM50-210730   Stool      2021-08-03       M50       粪便 CYM50-210735    2021.07.30     2021-08-06      7769
## S_7930 CYM50-210735-0727 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.27     2021-08-06      7770
## S_7931 CYM50-210735-0728 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.28     2021-08-06      7771
## S_7932 CYM50-210735-0729 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.29     2021-08-06      7772
## S_7933 CYM50-210735-0730 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.30     2021-08-06      7773
## S_7327         Community      QC            &lt;NA&gt;       Ref         QC         &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;      7222
##        SeqID_16s                                             Pipeline_MGS
## S_7929      7929 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
## S_7930      7930 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
## S_7931      7931 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
## S_7932      7932 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
## S_7933      7933 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
## S_7327      7327 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output
##                                                                Pipeline_16s
## S_7929 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7930 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7931 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7932 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7933 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7327 /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb97-1"><a href="test-example.html#cb97-1" aria-hidden="true" tabindex="-1"></a>run_RefCheck(</span>
<span id="cb97-2"><a href="test-example.html#cb97-2" aria-hidden="true" tabindex="-1"></a>    ps = dada2_ps_genus,</span>
<span id="cb97-3"><a href="test-example.html#cb97-3" aria-hidden="true" tabindex="-1"></a>    BRS_ID = &quot;S_7327&quot;,</span>
<span id="cb97-4"><a href="test-example.html#cb97-4" aria-hidden="true" tabindex="-1"></a>    Ref_type = &quot;16s&quot;)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for 16s
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [3]
## g__Lactobacillus
## g__Escherichia_Shigella
## g__Enterococcus
## The number of bacteria unmatched the Reference Matrix is [12]
## g__Bifidobacterium
## g__Bacteroides
## g__Faecalibacterium
## g__Parabacteroides
## g__Collinsella
## g__Coprococcus_3
## g__Dorea
## g__Streptococcus
## g__Roseburia
## g__Anaerostipes
## g__Prevotella_9
## g__Eggerthella
## The number of the additional bacteria compared to the Reference Matrix is [6]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: FALSE
## Correlation Coefficient of the BRS is: 6
## Bray Curtis of the BRS is: 0.2265
## Impurity of the BRS is: 68.36
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix
## ###########################################################
##                             8002        8003        8004        8005     8006      8007      8008      8009    S_7327
## Lactobacillus         2.61732573  3.36856272  3.44379163  3.88394343  5.92927  5.780008  5.781891  6.326723 12.263006
## Escherichia_Shigella 15.27581475 16.00265210 12.36527954 14.08077142 10.84235 13.432037 10.157432 11.599456 12.119942
## Enterococcus         14.51444842 14.66472707 11.04239952 11.66901114 13.16748 12.073609 12.887711 12.849247  7.258671
## Impurity_level        0.08562792  0.06531291  0.05987576  0.06228054  0.00000  0.000000  0.000000  0.000000 68.360000
##                          mean                                              Evaluation
## Lactobacillus         5.48828 S_7327 didn&#39;t pass the threshold (2022-08-09 09:46:07).
## Escherichia_Shigella 12.87508 S_7327 didn&#39;t pass the threshold (2022-08-09 09:46:07).
## Enterococcus         12.23637 S_7327 didn&#39;t pass the threshold (2022-08-09 09:46:07).
## Impurity_level        7.62590 S_7327 didn&#39;t pass the threshold (2022-08-09 09:46:07).</code></pre>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="test-example.html#cb99-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb99-2"><a href="test-example.html#cb99-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ps =</span> dada2_ps,</span>
<span id="cb99-3"><a href="test-example.html#cb99-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb99-4"><a href="test-example.html#cb99-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb99-5"><a href="test-example.html#cb99-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-6"><a href="test-example.html#cb99-6" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS</span>
<span id="cb99-7"><a href="test-example.html#cb99-7" aria-hidden="true" tabindex="-1"></a><span class="do">## phyloseq-class experiment-level object</span></span>
<span id="cb99-8"><a href="test-example.html#cb99-8" aria-hidden="true" tabindex="-1"></a><span class="do">## otu_table()   OTU Table:         [ 1948 taxa and 144 samples ]</span></span>
<span id="cb99-9"><a href="test-example.html#cb99-9" aria-hidden="true" tabindex="-1"></a><span class="do">## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]</span></span>
<span id="cb99-10"><a href="test-example.html#cb99-10" aria-hidden="true" tabindex="-1"></a><span class="do">## tax_table()   Taxonomy Table:    [ 1948 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb99-11"><a href="test-example.html#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="do">## phy_tree()    Phylogenetic Tree: [ 1948 tips and 1933 internal nodes ]</span></span>
<span id="cb99-12"><a href="test-example.html#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="do">## refseq()      DNAStringSet:      [ 1948 reference sequences ]</span></span>
<span id="cb99-13"><a href="test-example.html#cb99-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-14"><a href="test-example.html#cb99-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step3/&quot;</span>)) {</span>
<span id="cb99-15"><a href="test-example.html#cb99-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step3/&quot;</span>)</span>
<span id="cb99-16"><a href="test-example.html#cb99-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb99-17"><a href="test-example.html#cb99-17" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_remove_BRS, <span class="st">&quot;DataSet/Step3/Donor_16s_phyloseq_remove_BRS.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step4-rarefaction-curves" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Step4: Rarefaction curves<a href="test-example.html#step4-rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="test-example.html#cb100-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step3/Donor_16s_phyloseq_remove_BRS.RDS&quot;</span>)</span>
<span id="cb100-2"><a href="test-example.html#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="test-example.html#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RarefCurve</span>(<span class="at">ps =</span> dada2_ps_remove_BRS,</span>
<span id="cb100-4"><a href="test-example.html#cb100-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb100-5"><a href="test-example.html#cb100-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">step =</span> <span class="dv">400</span>,</span>
<span id="cb100-6"><a href="test-example.html#cb100-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb100-7"><a href="test-example.html#cb100-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## rarefying sample S_7271
## rarefying sample S_7272
## rarefying sample S_7273
## rarefying sample S_7274
## rarefying sample S_7275
## rarefying sample S_7276
## rarefying sample S_7277
## rarefying sample S_7278
## rarefying sample S_7279
## rarefying sample S_7280
## rarefying sample S_7281
## rarefying sample S_7282
## rarefying sample S_7283
## rarefying sample S_7284
## rarefying sample S_7285
## rarefying sample S_7286
## rarefying sample S_7287
## rarefying sample S_7288
## rarefying sample S_7289
## rarefying sample S_7290
## rarefying sample S_7291
## rarefying sample S_7292
## rarefying sample S_7293
## rarefying sample S_7294
## rarefying sample S_7295
## rarefying sample S_7296
## rarefying sample S_7297
## rarefying sample S_7298
## rarefying sample S_7299
## rarefying sample S_7300
## rarefying sample S_7301
## rarefying sample S_7302
## rarefying sample S_7303
## rarefying sample S_7304
## rarefying sample S_7305
## rarefying sample S_7306
## rarefying sample S_7307
## rarefying sample S_7308
## rarefying sample S_7309
## rarefying sample S_7310
## rarefying sample S_7311
## rarefying sample S_7312
## rarefying sample S_7313
## rarefying sample S_7314
## rarefying sample S_7315
## rarefying sample S_7316
## rarefying sample S_7317
## rarefying sample S_7318
## rarefying sample S_7319
## rarefying sample S_7320
## rarefying sample S_7321
## rarefying sample S_7322
## rarefying sample S_7323
## rarefying sample S_7324
## rarefying sample S_7325
## rarefying sample S_7326
## rarefying sample S_7846
## rarefying sample S_7847
## rarefying sample S_7848
## rarefying sample S_7849
## rarefying sample S_7850
## rarefying sample S_7851
## rarefying sample S_7852
## rarefying sample S_7853
## rarefying sample S_7854
## rarefying sample S_7855
## rarefying sample S_7856
## rarefying sample S_7857
## rarefying sample S_7858
## rarefying sample S_7859
## rarefying sample S_7860
## rarefying sample S_7861
## rarefying sample S_7862
## rarefying sample S_7863
## rarefying sample S_7864
## rarefying sample S_7865
## rarefying sample S_7866
## rarefying sample S_7867
## rarefying sample S_7868
## rarefying sample S_7869
## rarefying sample S_7870
## rarefying sample S_7871
## rarefying sample S_7872
## rarefying sample S_7873
## rarefying sample S_7874
## rarefying sample S_7875
## rarefying sample S_7876
## rarefying sample S_7877
## rarefying sample S_7878
## rarefying sample S_7879
## rarefying sample S_7880
## rarefying sample S_7881
## rarefying sample S_7882
## rarefying sample S_7883
## rarefying sample S_7884
## rarefying sample S_7885
## rarefying sample S_7886
## rarefying sample S_7887
## rarefying sample S_7888
## rarefying sample S_7889
## rarefying sample S_7890
## rarefying sample S_7891
## rarefying sample S_7892
## rarefying sample S_7893
## rarefying sample S_7894
## rarefying sample S_7895
## rarefying sample S_7896
## rarefying sample S_7897
## rarefying sample S_7898
## rarefying sample S_7899
## rarefying sample S_7900
## rarefying sample S_7901
## rarefying sample S_7902
## rarefying sample S_7903
## rarefying sample S_7904
## rarefying sample S_7905
## rarefying sample S_7906
## rarefying sample S_7907
## rarefying sample S_7908
## rarefying sample S_7909
## rarefying sample S_7910
## rarefying sample S_7911
## rarefying sample S_7912
## rarefying sample S_7913
## rarefying sample S_7914
## rarefying sample S_7915
## rarefying sample S_7916
## rarefying sample S_7917
## rarefying sample S_7918
## rarefying sample S_7919
## rarefying sample S_7920
## rarefying sample S_7921
## rarefying sample S_7922
## rarefying sample S_7923
## rarefying sample S_7924
## rarefying sample S_7925
## rarefying sample S_7926
## rarefying sample S_7927
## rarefying sample S_7928
## rarefying sample S_7929
## rarefying sample S_7930
## rarefying sample S_7931
## rarefying sample S_7932
## rarefying sample S_7933</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-62"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-62-1.png" alt="Rarefaction curves (Example)" width="768" />
<p class="caption">
Figure 9.2: Rarefaction curves (Example)
</p>
</div>
</div>
<div id="step5-rarefy-otu-counts" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> Step5: Rarefy otu counts<a href="test-example.html#step5-rarefy-otu-counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="test-example.html#cb102-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step3/Donor_16s_phyloseq_remove_BRS.RDS&quot;</span>)</span>
<span id="cb102-2"><a href="test-example.html#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> dada2_ps_remove_BRS)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 33267&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 153367&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 10909876&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 75763.0277777778&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 71985.5&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.912599104494638&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? YES&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 837&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 12&quot;
## 
## [[11]]
##  [1] &quot;SampleTubeID&quot;    &quot;Group&quot;           &quot;Date_Sequencing&quot; &quot;ProductID&quot;       &quot;SampleType&quot;      &quot;ProductBatch&quot;   
##  [7] &quot;Date_Sampling&quot;   &quot;Date_Receiving&quot;  &quot;SeqID_MGS&quot;       &quot;SeqID_16s&quot;       &quot;Pipeline_MGS&quot;    &quot;Pipeline_16s&quot;</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="test-example.html#cb104-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> dada2_ps_remove_BRS, </span>
<span id="cb104-2"><a href="test-example.html#cb104-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">size =</span> <span class="dv">33267</span>)</span>
<span id="cb104-3"><a href="test-example.html#cb104-3" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1091 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 1091 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1091 tips and 1086 internal nodes ]
## refseq()      DNAStringSet:      [ 1091 reference sequences ]</code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="test-example.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step5/&quot;</span>)) {</span>
<span id="cb106-2"><a href="test-example.html#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step5/&quot;</span>)</span>
<span id="cb106-3"><a href="test-example.html#cb106-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-4"><a href="test-example.html#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_rare, <span class="st">&quot;DataSet/Step5/Donor_16s_phyloseq_remove_BRS_rare.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step6-extracting-specific-taxonomic-level" class="section level2 hasAnchor" number="9.9">
<h2><span class="header-section-number">9.9</span> Step6: Extracting specific taxonomic level<a href="test-example.html#step6-extracting-specific-taxonomic-level" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="test-example.html#cb107-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step5/Donor_16s_phyloseq_remove_BRS_rare.RDS&quot;</span>)</span>
<span id="cb107-2"><a href="test-example.html#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="test-example.html#cb107-3" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rare, </span>
<span id="cb107-4"><a href="test-example.html#cb107-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb107-5"><a href="test-example.html#cb107-5" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 225 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 225 taxa by 6 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="test-example.html#cb109-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_order <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rare, </span>
<span id="cb109-2"><a href="test-example.html#cb109-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>)</span>
<span id="cb109-3"><a href="test-example.html#cb109-3" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_order</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 37 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 37 taxa by 4 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="test-example.html#cb111-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_phylum <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rare, </span>
<span id="cb111-2"><a href="test-example.html#cb111-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb111-3"><a href="test-example.html#cb111-3" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_phylum</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 15 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 15 taxa by 2 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="test-example.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step6/&quot;</span>)) {</span>
<span id="cb113-2"><a href="test-example.html#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step6/&quot;</span>)</span>
<span id="cb113-3"><a href="test-example.html#cb113-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb113-4"><a href="test-example.html#cb113-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_rare_genus, <span class="st">&quot;DataSet/Step6/Donor_16s_phyloseq_remove_BRS_rare_genus.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-5"><a href="test-example.html#cb113-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_rare_order, <span class="st">&quot;DataSet/Step6/Donor_16s_phyloseq_remove_BRS_rare_order.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-6"><a href="test-example.html#cb113-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_rare_phylum, <span class="st">&quot;DataSet/Step6/Donor_16s_phyloseq_remove_BRS_rare_phylum.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step7-globalview" class="section level2 hasAnchor" number="9.10">
<h2><span class="header-section-number">9.10</span> Step7: GlobalView<a href="test-example.html#step7-globalview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="test-example.html#cb114-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step6/Donor_16s_phyloseq_remove_BRS_rare_genus.RDS&quot;</span>)</span>
<span id="cb114-2"><a href="test-example.html#cb114-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-3"><a href="test-example.html#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="co"># alpha</span></span>
<span id="cb114-4"><a href="test-example.html#cb114-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb114-5"><a href="test-example.html#cb114-5" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb114-6"><a href="test-example.html#cb114-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dada2_ps_rare_genus_alpha,</span>
<span id="cb114-7"><a href="test-example.html#cb114-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb114-8"><a href="test-example.html#cb114-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb114-9"><a href="test-example.html#cb114-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb114-10"><a href="test-example.html#cb114-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-65-1"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-65-1.png" alt="diversity and ordination and composition(Example)" width="1152" />
<p class="caption">
Figure 9.3: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="test-example.html#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># beta</span></span>
<span id="cb115-2"><a href="test-example.html#cb115-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb115-3"><a href="test-example.html#cb115-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb115-4"><a href="test-example.html#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(<span class="at">datMatrix =</span> dada2_ps_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-65-2"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-65-2.png" alt="diversity and ordination and composition(Example)" width="1152" />
<p class="caption">
Figure 9.4: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="test-example.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># permanova</span></span>
<span id="cb116-2"><a href="test-example.html#cb116-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb116-3"><a href="test-example.html#cb116-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb116-4"><a href="test-example.html#cb116-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb116-5"><a href="test-example.html#cb116-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada2_ps_per)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs  MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## Group          144  1  1.335187 1.335187 9.669403 0.06375315  0.001          0.001</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="test-example.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># beta dispersion</span></span>
<span id="cb118-2"><a href="test-example.html#cb118-2" aria-hidden="true" tabindex="-1"></a>beta_df <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb118-3"><a href="test-example.html#cb118-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb118-4"><a href="test-example.html#cb118-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##            Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)  
## Groups      1 0.04375 0.043749 5.6785    999  0.016 *
## Residuals 142 1.09402 0.007704                       
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##          Product Stool
## Product          0.017
## Stool   0.018497</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="test-example.html#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ordination</span></span>
<span id="cb120-2"><a href="test-example.html#cb120-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb120-3"><a href="test-example.html#cb120-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ps =</span> dada2_ps_rare_genus,</span>
<span id="cb120-4"><a href="test-example.html#cb120-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb120-5"><a href="test-example.html#cb120-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb120-6"><a href="test-example.html#cb120-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-7"><a href="test-example.html#cb120-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> dada2_ps_ordination, </span>
<span id="cb120-8"><a href="test-example.html#cb120-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb120-9"><a href="test-example.html#cb120-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb120-10"><a href="test-example.html#cb120-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-65-3"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-65-3.png" alt="diversity and ordination and composition(Example)" width="1152" />
<p class="caption">
Figure 9.5: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="test-example.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Microbial composition</span></span>
<span id="cb121-2"><a href="test-example.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb121-3"><a href="test-example.html#cb121-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">phyloseq =</span> dada2_ps_rare_genus,</span>
<span id="cb121-4"><a href="test-example.html#cb121-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb121-5"><a href="test-example.html#cb121-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-65-4"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-65-4.png" alt="diversity and ordination and composition(Example)" width="1152" />
<p class="caption">
Figure 9.6: diversity and ordination and composition(Example)
</p>
</div>
</div>
<div id="step8-differential-analysis" class="section level2 hasAnchor" number="9.11">
<h2><span class="header-section-number">9.11</span> Step8: Differential Analysis<a href="test-example.html#step8-differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="test-example.html#cb122-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step6/Donor_16s_phyloseq_remove_BRS_rare_genus.RDS&quot;</span>)</span>
<span id="cb122-2"><a href="test-example.html#cb122-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-3"><a href="test-example.html#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="co"># filter &amp; trim</span></span>
<span id="cb122-4"><a href="test-example.html#cb122-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb122-5"><a href="test-example.html#cb122-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">cutoff =</span> <span class="dv">10</span>, </span>
<span id="cb122-6"><a href="test-example.html#cb122-6" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb122-7"><a href="test-example.html#cb122-7" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> dada2_ps_rare_genus_filter, </span>
<span id="cb122-8"><a href="test-example.html#cb122-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb122-9"><a href="test-example.html#cb122-9" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb122-10"><a href="test-example.html#cb122-10" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 127 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 127 taxa by 6 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="test-example.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lefse</span></span>
<span id="cb124-2"><a href="test-example.html#cb124-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb124-3"><a href="test-example.html#cb124-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb124-4"><a href="test-example.html#cb124-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb124-5"><a href="test-example.html#cb124-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb124-6"><a href="test-example.html#cb124-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb124-7"><a href="test-example.html#cb124-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Lda =</span> <span class="dv">2</span>)</span>
<span id="cb124-8"><a href="test-example.html#cb124-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-9"><a href="test-example.html#cb124-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb124-10"><a href="test-example.html#cb124-10" aria-hidden="true" tabindex="-1"></a><span class="co"># dada2_ps_lefse$LDA_Score &lt;- dada2_ps_lefse$LDA_Score * 1000</span></span>
<span id="cb124-11"><a href="test-example.html#cb124-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-12"><a href="test-example.html#cb124-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb124-13"><a href="test-example.html#cb124-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> dada2_ps_lefse,</span>
<span id="cb124-14"><a href="test-example.html#cb124-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb124-15"><a href="test-example.html#cb124-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb124-16"><a href="test-example.html#cb124-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-66-1"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-66-1.png" alt="Differential Analysis (Example)" width="768" />
<p class="caption">
Figure 9.7: Differential Analysis (Example)
</p>
</div>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="test-example.html#cb125-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb125-2"><a href="test-example.html#cb125-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb125-3"><a href="test-example.html#cb125-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb125-4"><a href="test-example.html#cb125-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>))</span>
<span id="cb125-5"><a href="test-example.html#cb125-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb125-6"><a href="test-example.html#cb125-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> dada2_ps_wilcox,</span>
<span id="cb125-7"><a href="test-example.html#cb125-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb125-8"><a href="test-example.html#cb125-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">Stool_vs_Product&quot;</span>,</span>
<span id="cb125-9"><a href="test-example.html#cb125-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb125-10"><a href="test-example.html#cb125-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb125-11"><a href="test-example.html#cb125-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-12"><a href="test-example.html#cb125-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb125-13"><a href="test-example.html#cb125-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-66-2"></span>
<img src="16s-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-66-2.png" alt="Differential Analysis (Example)" width="768" />
<p class="caption">
Figure 9.8: Differential Analysis (Example)
</p>
</div>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="test-example.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step8/&quot;</span>)) {</span>
<span id="cb126-2"><a href="test-example.html#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step8/&quot;</span>)</span>
<span id="cb126-3"><a href="test-example.html#cb126-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb126-4"><a href="test-example.html#cb126-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dada2_ps_rare_genus_filter_trim, <span class="st">&quot;DataSet/Step8/Donor_16s_phyloseq_remove_BRS_rare_genus_filter_trim.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="systematic-information-7" class="section level2 hasAnchor" number="9.12">
<h2><span class="header-section-number">9.12</span> Systematic Information<a href="test-example.html#systematic-information-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="test-example.html#cb127-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-09
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18     2021-09-16 [1] CRAN (R 4.1.0)
##  ALDEx2                 1.26.0     2021-10-26 [1] Bioconductor
##  annotate               1.72.0     2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2     2021-11-09 [1] Bioconductor
##  ape                    5.6-2      2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  biomformat             1.22.0     2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2      2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27       2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12     2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  Cairo                  1.5-14     2021-12-21 [1] CRAN (R 4.1.2)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12     2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5      2022-01-06 [1] CRAN (R 4.1.2)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [1] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0      2020-02-06 [1] CRAN (R 4.1.0)
##  circlize               0.4.14     2022-02-11 [1] CRAN (R 4.1.2)
##  class                  7.3-20     2022-01-13 [1] CRAN (R 4.1.2)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.1.2)
##  clue                   0.3-60     2021-10-11 [1] CRAN (R 4.1.0)
##  cluster              * 2.1.2      2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [1] Bioconductor
##  conflicted             1.1.0      2021-11-26 [1] CRAN (R 4.1.0)
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.1.0)
##  corrplot               0.92       2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0      2022-02-14 [1] CRAN (R 4.1.2)
##  crmn                   0.0.21     2020-02-10 [1] CRAN (R 4.1.0)
##  curl                   4.3.2      2021-06-23 [1] CRAN (R 4.1.0)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  dendextend           * 1.15.2     2021-10-28 [1] CRAN (R 4.1.0)
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [1] Bioconductor
##  devtools               2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.1.2)
##  doSNOW                 1.0.20     2022-02-04 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8      2022-02-08 [1] CRAN (R 4.1.2)
##  dynamicTreeCut       * 1.63-1     2016-03-11 [1] CRAN (R 4.1.0)
##  e1071                  1.7-9      2021-09-16 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0     2021-10-26 [1] Bioconductor
##  ellipse                0.4.2      2020-05-27 [1] CRAN (R 4.1.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastcluster          * 1.2.3      2021-05-24 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  fastmatch              1.1-3      2021-07-23 [1] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [1] CRAN (R 4.1.0)
##  fgsea                  1.20.0     2021-10-26 [1] Bioconductor
##  filematrix             1.3        2018-02-27 [1] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82     2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1      2021-09-03 [1] CRAN (R 4.1.0)
##  Formula                1.2-4      2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
##  future                 1.24.0     2022-02-19 [1] CRAN (R 4.1.2)
##  future.apply           1.8.1      2021-08-10 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0     2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [1] Bioconductor
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.1.0)
##  ggforce                0.3.3      2021-03-05 [1] CRAN (R 4.1.0)
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0      2020-06-27 [1] CRAN (R 4.1.0)
##  ggraph               * 2.0.5      2021-02-23 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3      2021-09-09 [1] CRAN (R 4.1.0)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet               * 4.1-3      2021-11-02 [1] CRAN (R 4.1.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.1.0)
##  globals                0.14.0     2020-11-22 [1] CRAN (R 4.1.0)
##  globaltest             5.48.0     2021-10-26 [1] Bioconductor
##  glue                 * 1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [1] Bioconductor
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1      2020-11-28 [1] CRAN (R 4.1.0)
##  graphlayouts           0.8.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2      2021-06-06 [1] CRAN (R 4.1.0)
##  hardhat                0.2.0      2022-01-24 [1] CRAN (R 4.1.2)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0      2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0      2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.1.0)
##  httr                 * 1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  huge                   1.3.5      2021-06-30 [1] CRAN (R 4.1.0)
##  igraph               * 1.2.11     2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0     2021-10-26 [1] Bioconductor
##  ipred                  0.9-12     2021-09-15 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  irlba                  2.3.5      2021-12-06 [1] CRAN (R 4.1.0)
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9      2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0     2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20    2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39       2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 4.1.0)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.1.0)
##  lavaan                 0.6-11     2022-03-31 [1] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1     2022-02-17 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.1.0)
##  locfit                 1.5-9.5    2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55     2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix               * 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  mbzinb                 0.2        2022-03-16 [1] local
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  MetaboAnalystR       * 3.2.0      2022-06-28 [1] Github (xia-lab/MetaboAnalystR@892a31b)
##  metagenomeSeq          1.36.0     2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39     2022-02-24 [1] CRAN (R 4.1.2)
##  mixedCCA               1.5.2      2022-07-14 [1] Github (irinagain/mixedCCA@c6d41a3)
##  mixOmics               6.18.1     2021-11-18 [1] Bioconductor (R 4.1.2)
##  mnormt                 2.0.2      2020-09-01 [1] CRAN (R 4.1.0)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18     2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0     2021-10-26 [1] Bioconductor
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1    2020-03-22 [1] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3      2022-07-14 [1] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-155    2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17     2022-01-13 [1] CRAN (R 4.1.2)
##  parallelly             1.31.0     2022-04-07 [1] CRAN (R 4.1.2)
##  pbapply                1.5-0      2021-09-16 [1] CRAN (R 4.1.0)
##  pbivnorm               0.6.0      2015-01-23 [1] CRAN (R 4.1.0)
##  pcaMethods             1.86.0     2021-10-26 [1] Bioconductor
##  pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 4.1.0)
##  permute                0.9-7      2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12     2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0     2021-10-26 [1] Bioconductor
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.1.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.1.0)
##  POMA                 * 1.7.2      2022-07-26 [1] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [1] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.1.0)
##  processx               3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26     2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
##  pscl                   1.5.5      2020-03-07 [1] CRAN (R 4.1.0)
##  psych                  2.2.5      2022-05-10 [1] CRAN (R 4.1.2)
##  pulsar                 0.3.7      2020-08-07 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  qgraph                 1.9.2      2022-03-04 [1] CRAN (R 4.1.2)
##  qs                     0.25.3     2022-02-22 [1] CRAN (R 4.1.2)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  ragg                   1.2.2      2022-02-21 [1] CRAN (R 4.1.2)
##  randomForest           4.7-1      2022-02-03 [1] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [1] Bioconductor
##  RApiSerialize          0.1.0      2014-04-19 [1] CRAN (R 4.1.0)
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.1.0)
##  rbibutils              2.2.7      2021-12-07 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2    2022-03-11 [1] CRAN (R 4.1.2)
##  RcppParallel           5.1.5      2022-01-05 [1] CRAN (R 4.1.2)
##  RcppZiggurat           0.1.6      2020-10-20 [1] CRAN (R 4.1.0)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.1.2)
##  Rdpack                 2.2        2022-03-19 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.1.2)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.1.0)
##  Rfast                  2.0.6      2022-02-16 [1] CRAN (R 4.1.2)
##  rhdf5                  2.38.1     2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [1] Bioconductor
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14       2022-04-25 [1] CRAN (R 4.1.2)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.1.2)
##  rootSolve              1.8.2.3    2021-09-29 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  Rserve               * 1.8-10     2021-11-25 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0     2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10     2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0      2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sandwich               3.0-1      2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0      2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  scrime                 1.3.5      2018-12-01 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.1.0)
##  siggenes               1.68.0     2021-10-26 [1] Bioconductor
##  snow                   0.4-4      2021-10-27 [1] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringfish             0.15.5     2021-12-01 [1] CRAN (R 4.1.0)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4      2022-02-11 [1] CRAN (R 4.1.2)
##  testthat               3.1.2      2022-01-20 [1] CRAN (R 4.1.2)
##  textshaping            0.3.6      2021-10-13 [1] CRAN (R 4.1.0)
##  TH.data                1.1-0      2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidygraph              1.2.1      2022-04-05 [1] CRAN (R 4.1.2)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.1.2)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.1.0)
##  tmvnsim                1.0-2      2016-12-15 [1] CRAN (R 4.1.0)
##  truncnorm              1.0-8      2018-02-27 [1] CRAN (R 4.1.0)
##  tweenr                 1.0.2      2021-03-23 [1] CRAN (R 4.1.0)
##  usethis                2.1.5      2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7      2020-11-28 [1] CRAN (R 4.1.0)
##  VGAM                   1.1-6      2022-02-14 [1] CRAN (R 4.1.2)
##  viridis                0.6.2      2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0      2021-04-13 [1] CRAN (R 4.1.0)
##  WGCNA                * 1.71       2022-04-22 [1] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [1] Bioconductor
##  xfun                   0.30       2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.7.3    2022-08-05 [1] local
##  XML                    3.99-0.9   2022-02-24 [1] CRAN (R 4.1.2)
##  xtable                 1.8-4      2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0      2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
##  zoo                    1.8-9      2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/16s_DataAnalysis_template/blob/main/08-Example.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["16s microbiota data analysis based on XMAS package.pdf", "16s microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
